core/src/serde_support.rs

```diff
@@ -0,0 +1,589 @@
+                        }
+                        "pos" => {
+                            if pos.is_some() {
+                                return Err(SerdeError::duplicate_field("pos"));
+                            }
+                            pos = Some(map.next_value()?);
+                        }
+                        "value_enc" => {
+                            if sender.is_some() {
+                                return Err(SerdeError::duplicate_field(
+                                    "value_enc",
+                                ));
+                            }
+                            value_enc = Some(map.next_value()?);
+                        }
+                        "sender" => {
+                            if sender.is_some() {
+                                return Err(SerdeError::duplicate_field(
+                                    "sender",
+                                ));
+                            }
+                            sender = Some(map.next_value()?);
+                        }
+                        field => {
+                            return Err(SerdeError::unknown_field(
+                                field, &FIELDS,
+                            ))
+                        }
+                    }
+                }
+
+                let value_enc: String = value_enc
+                    .ok_or_else(|| SerdeError::missing_field("value_enc"))?;
+                let value_enc_len = value_enc.len();
+                let decoded_value_enc =
+                    hex::decode(value_enc).map_err(SerdeError::custom)?;
+                let value_enc: [u8; NOTE_VAL_ENC_SIZE] =
+                    decoded_value_enc.try_into().map_err(|_| {
+                        SerdeError::invalid_length(
+                            value_enc_len,
+                            &NOTE_VAL_ENC_SIZE.to_string().as_str(),
+                        )
+                    })?;
+
+                Ok(Note {
+                    note_type: note_type.ok_or_else(|| {
+                        SerdeError::missing_field("note_type")
+                    })?,
+                    stealth_address: stealth_address.ok_or_else(|| {
+                        SerdeError::missing_field("stealth_address")
+                    })?,
+                    value_commitment: value_commitment.ok_or_else(|| {
+                        SerdeError::missing_field("value_commitment")
+                    })?,
+                    pos: pos.ok_or_else(|| SerdeError::missing_field("pos"))?.0,
+                    value_enc,
+                    sender: sender
+                        .ok_or_else(|| SerdeError::missing_field("sender"))?,
+                })
+            }
+        }
+
+        deserializer.deserialize_struct("Note", &FIELDS, NoteVisitor)
+    }
+}
+
+// The current serde implementation for `BlsScalar` is not what it's expected to
+// be, so this is needed at the moment.
```

Comment:
Could you add the issue: https://github.com/dusk-network/bls12_381/issues/145 to this comment as well?